packages/running/src/index.tsx

```diff
@@ -250,6 +251,90 @@ function List(props: {
   );
 }
 
+class ListWidget extends ReactWidget {
+  constructor(
+    private _options: {
+      manager: IRunningSessions.IManager;
+      runningItems: IRunningSessions.IRunningItem[];
+      shutdownAllLabel: string;
+      translator?: ITranslator;
+    }
+  ) {
+    super();
+    _options.manager.runningChanged.connect(this._emitUpdate, this);
+  }
+
+  dispose() {
+    super.dispose();
+    this._options.manager.runningChanged.disconnect(this._emitUpdate, this);
```

Comment:
The undo action should mirror the action order.

```suggestion
    this._options.manager.runningChanged.disconnect(this._emitUpdate, this);
    super.dispose();
```